### PR TITLE
[MUI-536]: Create Date Input Component

### PR DESCRIPTION
**Pull request by mr-teddy-bear**
# Description

Create Date Input Component

## Jira ticket(optional)

[MUI-536](https://minterestfinance.atlassian.net/jira/software/projects/MUI/boards/7?assignee=62591503eee0a9006ab98b74&selectedIssue=MUI-536)

## Type of change

Please pick options that are relevant.

- [ ] Bug fix OR refactoring (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature or refactoring that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/components/DatePicker/DatePicker.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+}));
+
+export const CustomDatePicker = ({
+  labelTitle,
+  ...props
+}: CustomProps & DatePickerProps<any, any>): JSX.Element => {
+  return (
+    <LocalizationProvider dateAdapter={AdapterDateFns}>
+      <DatePicker
+        {...props}
```
Comment (gavrilin-remote) [inline src/components/DatePicker/DatePicker.tsx:68]: place rest after all predefined props, to make this component flexible

File: src/components/DatePicker/DatePicker.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  borderRadius: '4px',
+  height: 40,
+  svg: {
+    color: 'rgba(109,118,146,0.4)',
+    width: '18px',
+    height: '18px',
+  },
+}));
+
+const Input = styled('input')(() => ({
```
Comment (gavrilin-remote) [inline src/components/DatePicker/DatePicker.tsx:46]: background color in input is different than in Wrapper 
Снимок экрана 2022-09-02 в 13 41 41

Comment (gavrilin-remote) [reply src/components/DatePicker/DatePicker.tsx:46]: input more white

Comment (mr-teddy-bear) [reply src/components/DatePicker/DatePicker.tsx:46]: Done

File: src/components/DatePicker/DatePicker.tsx (hunk 1 above)
Comment (mr-teddy-bear) [reply src/components/DatePicker/DatePicker.tsx]: Done

File: src/components/DatePicker/DatePicker.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+import { styled } from '@mui/material';
+import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
+import { DatePicker, DatePickerProps } from '@mui/x-date-pickers/DatePicker';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+
+import { CalendarIcon } from 'assets/svg';
+
+type CustomProps = {
+  labelTitle: string;
```
Comment (gavrilin-remote) [inline src/components/DatePicker/DatePicker.tsx:11]: pick one
```suggestion
  label: string;
```
```suggestion
  title: string;
```

File: src/components/DatePicker/DatePicker.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  return (
+    <LocalizationProvider dateAdapter={AdapterDateFns}>
+      <DatePicker
+        label={labelTitle}
+        components={{
+          OpenPickerIcon: CalendarIcon,
+        }}
+        inputFormat='dd.MM.yyy'
+        {...props}
+        value={props.value}
```
Comment (gavrilin-remote) [inline src/components/DatePicker/DatePicker.tsx:75]: remove this line, rest operator already pass `value`

File: src/components/DatePicker/DatePicker.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        onChange={props.onChange}
```
Comment (gavrilin-remote) [inline src/components/DatePicker/DatePicker.tsx:76]: remove this line, rest operator already pass `value`

File: src/components/DatePicker/DatePicker.tsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+        renderInput={({ inputRef, inputProps, InputProps }) => (
+          <Wrapper>
+            {!!labelTitle && <Label>{labelTitle}</Label>}
+            <InputWrapper>
+              {InputProps?.endAdornment}
+
+              <Input ref={inputRef} {...inputProps} placeholder='DD.MM.YYYY' />
```
Comment (gavrilin-remote) [inline src/components/DatePicker/DatePicker.tsx:83]:
```suggestion
              <Input ref={inputRef}  placeholder='DD.MM.YYYY' {...inputProps} />
```

File: src/components/DatePicker/DatePicker.tsx [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+        components={{
+          OpenPickerIcon: CalendarIcon,
+        }}
+        inputFormat='dd.MM.yyy'
+        {...props}
+        value={props.value}
+        onChange={props.onChange}
+        renderInput={({ inputRef, inputProps, InputProps }) => (
+          <Wrapper>
+            {!!labelTitle && <Label>{labelTitle}</Label>}
```
Comment (gavrilin-remote) [inline src/components/DatePicker/DatePicker.tsx:79]: this will render Label even if text === ""

Comment (mr-teddy-bear) [reply src/components/DatePicker/DatePicker.tsx:79]: !!"" === false

File: src/components/DatePicker/DatePicker.tsx (hunk 6 above)
Comment (mr-teddy-bear) [reply src/components/DatePicker/DatePicker.tsx]: Done

File: src/components/DatePicker/DatePicker.tsx (hunk 5 above)
Comment (mr-teddy-bear) [reply src/components/DatePicker/DatePicker.tsx]: Done

File: src/components/DatePicker/DatePicker.tsx (hunk 4 above)
Comment (mr-teddy-bear) [reply src/components/DatePicker/DatePicker.tsx]: Done

File: src/components/DatePicker/DatePicker.tsx (hunk 3 above)
Comment (mr-teddy-bear) [reply src/components/DatePicker/DatePicker.tsx]: Done

File: src/components/DatePicker/DatePicker.tsx [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+export const CustomDatePicker = ({
+  label,
+  ...props
+}: CustomProps & DatePickerProps<any, any>): JSX.Element => {
+  return (
+    <LocalizationProvider dateAdapter={AdapterDateFns}>
+      <DatePicker
+        label={label}
+        components={{
+          OpenPickerIcon: CalendarIcon,
```
Comment (gavrilin-remote) [inline src/components/DatePicker/DatePicker.tsx:71]: Снимок экрана 2022-09-06 в 11 58 42
the icon is adjusted to right, can we place it to center?

Comment (mr-teddy-bear) [reply src/components/DatePicker/DatePicker.tsx:83]: Done

**Review by gavrilin-remote — APPROVED** [review]

**Pull request merged**